File: app/src/androidTest/java/io/github/fvasco/pinpoi/importer/AbstractImporterTestCase.java
package io.github.fvasco.pinpoi.importer;

import android.test.AndroidTestCase;

import java.io.InputStream;
import java.util.List;

import io.github.fvasco.pinpoi.model.Placemark;

/**
 * Base importer test case class.
 *
 * @author Francesco Vasco
 */
public abstract class AbstractImporterTestCase extends AndroidTestCase {

    /**
     * Execute import
     *
     * @param resource resource file name
     * @return list of imported placemark
     */
    public List<Placemark> importPlacemark(final AbstractImporter importer, final String resource) throws Exception {
        final InputStream is = getClass().getResourceAsStream(resource);
        try {
            return importPlacemark(importer, is);
        } finally {
            is.close();
        }
    }

    /**
     * Execute import
     */
    public List<Placemark> importPlacemark(final AbstractImporter importer, final InputStream is) throws Exception {
        final ListConsumer<Placemark> list = new ListConsumer<>();
        importer.setCollectionId(1);
        importer.setConsumer(list);
        importer.importPlacemarks(is);
        for (final Placemark p : list) {
            assertEquals(0, p.getId());
            assertNotNull(p.getName());
            assertTrue(!p.getName().isEmpty());
            assertTrue(!Float.isNaN(p.getLatitude()));
            assertTrue(!Float.isNaN(p.getLongitude()));
            assertEquals(1, p.getCollectionId());
        }
        return list;
    }
}

File: app/src/androidTest/java/io/github/fvasco/pinpoi/importer/AbstractImporterTestCase.kt
package io.github.fvasco.pinpoi.importer

import android.content.Context
import androidx.test.platform.app.InstrumentationRegistry
import io.github.fvasco.pinpoi.model.Placemark
import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Before
import java.io.InputStream
import java.util.*


/**
 * Base importer test case class.

 * @author Francesco Vasco
 */
abstract class AbstractImporterTestCase {


    protected lateinit var context: Context
        private set

    @Before
    fun setup() {
        context = InstrumentationRegistry.getContext()
    }

    /**
     * Execute import

     * @param resource resource file name
     * *
     * @return list of imported placemark
     */
    @Throws(Exception::class)
    fun importPlacemark(importer: AbstractImporter, resource: String, fileFormatFilter: FileFormatFilter = FileFormatFilter.NONE): List<Placemark> {
        javaClass.getResourceAsStream(resource).use {
            return importPlacemark(importer, it, fileFormatFilter)
        }
    }

    /**
     * Execute import
     */
    @Throws(Exception::class)
    fun importPlacemark(importer: AbstractImporter, input: InputStream, fileFormatFilter: FileFormatFilter = FileFormatFilter.NONE): List<Placemark> {
        val list = ArrayList<Placemark>()
        importer.collectionId = 1
        importer.consumer = { list.add(it) }
        importer.fileFormatFilter = fileFormatFilter
        importer.importPlacemarks(input)
        for (p in list) {
            assertEquals(0, p.id)
            assertTrue(!p.name.isEmpty())
            assertTrue(!p.coordinates.latitude.isNaN())
            assertTrue(!p.coordinates.longitude.isNaN())
            assertEquals(1, p.collectionId)
        }
        return list
    }
}
